### PR TITLE
public API v1

### PR DESCRIPTION
**Pull request by dspasibenko**
The very basic public API. Let's start from there and then will see...

### CODE REVIEW

File: api/proto/solaris/v1/solaris.proto [hunk 1]
```diff
@@ -1,5 +1,116 @@
...
+  rpc QueryRecords(QueryRecordsRequest) returns (QueryRecordsResult);
+}
+
+// Record represents one record of a log
+message Record {
+  // id is the record unique identifier. A record ID is ULID, so it is soreted and globaly unique.
+  string id = 1;
+  // logID is the log identifier the record is associated with. In a bunch result the field may be empty,
+  // this case the logID is same as for the previous record in the batch
+  optional string logID = 2;
```
Comment (kbabushkin) [inline api/proto/solaris/v1/solaris.proto:32]: nit: I would specify `optional` for only when it is optional in request specifically. I think it is not true for `logID`.

File: api/proto/solaris/v1/solaris.proto [hunk 2]
```diff
@@ -1,5 +1,116 @@
...
+  // AppendRecords appends a bunch of records to the log
+  rpc AppendRecords(AppendRecordsRequest) returns (Log);
+  // QueryRecords read records from one or many logs, merging them together into the result set
+  // sorted in ascending or descending order by the records IDs (timestamps)
+  rpc QueryRecords(QueryRecordsRequest) returns (QueryRecordsResult);
+}
+
+// Record represents one record of a log
+message Record {
+  // id is the record unique identifier. A record ID is ULID, so it is soreted and globaly unique.
```
Comment (kbabushkin) [inline api/proto/solaris/v1/solaris.proto:28]: nit: Not sure if we need to say that it is ULID specifically

File: api/proto/solaris/v1/solaris.proto (hunk 1 above)
Comment (dspasibenko) [reply api/proto/solaris/v1/solaris.proto:32]: The field may be empty to save 16 bytes :)

File: api/proto/solaris/v1/solaris.proto [hunk 3]
```diff
@@ -1,5 +1,116 @@
...
+  // logsCondition allows to specify the filter condition for selecting logs.
+  string logsCondition = 1;
+  // condition allows to specify the filter for the records.
+  string condition = 2;
+  // logIDs allows to specify the list of logs explicitly. If it is provided, then the logsCondition will be ignored.
+  repeated string logIDs = 3;
+  // descending specifies that the result should be sorted in the descending order
+  bool descending = 4;
+  // pageID provides the page ID if pagination is required
+  string pageID = 5;
```
[the remark this answers is not in the document]
Comment (dspasibenko) [reply api/proto/solaris/v1/solaris.proto:103]: yeah

File: api/proto/solaris/v1/solaris.proto (hunk 2 above)
Comment (dspasibenko) [reply api/proto/solaris/v1/solaris.proto:28]: well, ok, will rephrase

**Pull request merged**